router/peer.go

```diff
@@ -17,11 +17,24 @@ func ParsePeerUID(s string) (PeerUID, error) {
 	return PeerUID(uid), err
 }
 
+// Short IDs exist for the sake of fast datapath.  They are 12 bits,
+// randomly assigned, but we detect and recover from collisions.  This
+// does limit us to 4096 peers, but that should be sufficient for a
+// while.
+type PeerShortID uint16
+
+const PeerShortIDBits = 12
+
+func randomPeerShortID() PeerShortID {
+	return PeerShortID(randUint64() & (1<<PeerShortIDBits - 1))
```

Comment:
* Later on in the FDP commit the hexadecimal literal `0xfff` is used. Perhaps introduce a `MaxPeerShortID` constant initialised to `1<<PeerShortIDBits - 1` and use it here and there?
* Rather trivial, but `randUint64` reads eight bytes from the CSPRNG and then discards > 3/4 of the bits. We could add `randUint16` to `router/utils.go`